pkg/resources/validation/podsecurityadmissionconfigurationtemplate/podsecurityact.go

```diff
@@ -0,0 +1,299 @@
+
+// NewValidator returns a validator for PodSecurityAdmissionConfigurationTemplates
+func NewValidator(managementCache v3.ClusterCache, provisioningCache v1.ClusterCache) *Validator {
+	val := &Validator{
+		ManagementClusterCache:   managementCache,
+		provisioningClusterCache: provisioningCache,
+	}
+	val.ManagementClusterCache.AddIndexer(byPodSecurityAdmissionConfigurationName, byPodSecurityAdmissionConfigurationTemplateV3)
+	val.provisioningClusterCache.AddIndexer(byPodSecurityAdmissionConfigurationName, byPodSecurityAdmissionConfigurationTemplateV1)
+	return val
+}
+
+// ValidatingWebhook returns the ValidatingWebhook used for this CRD.
+func (v *Validator) ValidatingWebhook(clientConfig admissionregistrationv1.WebhookClientConfig) *admissionregistrationv1.ValidatingWebhook {
+	valWebhook := admission.NewDefaultValidatingWebhook(v, clientConfig, admissionregistrationv1.AllScopes)
+	valWebhook.FailurePolicy = admission.Ptr(admissionregistrationv1.Ignore)
+	return valWebhook
+}
+
+func byPodSecurityAdmissionConfigurationTemplateV1(obj *provv1.Cluster) ([]string, error) {
+	if obj.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName == "" {
+		return nil, nil
+	}
+	return []string{obj.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName}, nil
+}
+
+func byPodSecurityAdmissionConfigurationTemplateV3(obj *mgmtv3.Cluster) ([]string, error) {
+	if obj.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName == "" {
+		return nil, nil
+	}
+	return []string{obj.Spec.DefaultPodSecurityAdmissionConfigurationTemplateName}, nil
+}
+
+// GVR returns the GroupVersionKind for this CRD.
+func (v *Validator) GVR() schema.GroupVersionResource {
+	return gvr
+}
+
+// Operations returns list of operations handled by this validator.
+func (v *Validator) Operations() []admissionregistrationv1.OperationType {
+	return []admissionregistrationv1.OperationType{admissionregistrationv1.Update, admissionregistrationv1.Create, admissionregistrationv1.Delete}
+}
+
+// Admit handles the webhook admission request sent to this webhook.
+func (v *Validator) Admit(req *admission.Request) (*admissionv1.AdmissionResponse, error) {
+	listTrace := trace.New("PodSecurityAdmissionConfigurationTemplate Admit", trace.Field{Key: "user", Value: req.UserInfo.Username})
+	defer listTrace.LogIfLong(2 * time.Second)
+
+	resp := &admissionv1.AdmissionResponse{}
+	oldTemplate, newTemplate, err := objectsv3.PodSecurityAdmissionConfigurationTemplateOldAndNewFromRequest(&req.AdmissionRequest)
+	if err != nil {
+		return resp, fmt.Errorf("failed to parse PodSecurityAdmissionConfigurationTemplate object from request:%w", err)
+	}
+
+	switch req.Operation {
+	case admissionv1.Create, admissionv1.Update:
+		err = v.validateConfiguration(newTemplate)
+		if err != nil {
+			resp.Result = &metav1.Status{
+				Status:  "Failure",
+				Message: err.Error(),
+				Reason:  metav1.StatusReasonBadRequest,
+				Code:    http.StatusUnprocessableEntity,
+			}
+			resp.Allowed = false
+			break
```

Comment:
nit: for some of these cases it appears we can just return early instead of breaking.
```suggestion
			return resp, nil
```